### DISCUSSION TITLE
Custom operator for when a k8s namespace is being deleted (in Terminating State)

### DISCUSSION THREAD

**Discussion by gojanpaolo** [dotnet/dotnet-operator-sdk#495] [Q&A] [answered]
Hello, is there a suggested way to write a custom operator for when a k8s namespace is being deleted (in Terminating State)?

Our use case is that we need to run the solution as described below whenever a namespace is being deleted

> Issue: Unable to delete Kubernetes resources
> Kubernetes resources cannot be deleted, for example, if CFK pod gets deleted before other resources, if CFK can’t delete resources, or if the namespace is in termination state.
> Solution: Remove the finalizer using the following command:

```
kubectl get <resource> --no-headers | \
  awk '{print $1 }' | \
  xargs kubectl patch <resource> -p '{"metadata":{"finalizers":[]}}' \
  --type=merge
```
Reference: https://docs.confluent.io/operator/current/co-troubleshooting.html#issue-unable-to-delete-kubernetes-resources

---

I think we'll start trying the following: https://buehler.github.io/dotnet-operator-sdk/docs/entities.html#ignoring-entities

```
public class NamespaceSpec
{
    public string Name { get; set; }
}

[IgnoreEntity]
[KubernetesEntity(Group = "", ApiVersion = "v1")]
public class Namespace : CustomKubernetesEntity<NamespaceSpec>
{
}
```

**Comment by buehler** [comment, marked as answer]
Hey @gojanpaolo 

The normal namespace resource is already defined in the Kubernetes library.
You can use the `V1Namespace` (IIRC it is called so) and then create a controller for that resource:
```csharp
public class NsCtrl : IResourceController<V1Namespace>
{
}
```

**Reply by gojanpaolo** [reply]
@buehler Thanks for the immediate response! We'll try the built-in class.
Btw, do you think implementing a custom operator for our use case mentioned above is advisable? I feel there might be better solutions. TIA!

**Reply by buehler** [reply]
Theoretically an operator (and the operator pattern) can be used for any use-cases you seem fit. You can certainly use the controller to fetch all resources and objects in a namespace and remove any finalizers in them. But be advised, that you may remove finalizer that are required from Kubernets/other operators.

**Reply by gojanpaolo** [reply]
Got it! Thank you for the answer and developing this library!
